pkg/receive/handler.go

```diff
@@ -1273,20 +1302,32 @@ type peerGroup struct {
 	dialer func(ctx context.Context, target string, opts ...grpc.DialOption) (conn *grpc.ClientConn, err error)
 }
 
+func (p *peerGroup) closeAll() error {
+	p.m.Lock()
+	defer p.m.Unlock()
+	for addr := range p.connections {
+		if err := p.closeUnlocked(addr); err != nil {
+			return err
+		}
```

Comment:
Should it remember the error, but continue and try to close all connections?

Reply:
yep, we should do this.